### PR TITLE
Add api changes as of october

### PR DESCRIPTION
**Pull request by bomjacob**
Add changes as of October 3, 2016.
Mostly games: https://core.telegram.org/bots/api#games
Other changes: https://core.telegram.org/bots/api#recent-changes

- [x] Games
  - [ ] Unitests
- [x] `getWebhookInfo` and `WebhookInfo`
  - [x] Unitests
- [x] Allow HTTP url in all methods that accept `InputFile` or `file_id` (this requires disabling our own handling of this)
  - [x] Unitests
- [x] Url parameter in `answerCallbackQuery`
  - [ ] Unitests
- [x] `switch_inline_query_current_chat`
  - [ ] Unitests
- [x] Add caption fields to `send_audio`, `send_voice`, `InlineQueryResultAudio`, `InlineQueryResultVoice`, ``InlineQueryResultCachedAudio` and `InlineQueryResultCachedVoice`
  - [x] Unitests
- [x] `all_members_are_admins` in `Chat`
  - [x] Unitests
- [x] Handle the errors of `ResponseParameters` (no need unitests for the moment. we already had one response param and we didn't unitest it)

### CODE REVIEW

File: telegram/bot.py [hunk 1]
```diff
@@ -728,6 +743,47 @@ def sendContact(self, chat_id, phone_number, first_name, last_name=None, **kwarg
 
     @log
     @message
+    def sendGame(self,
+                 chat_id,
+                 game_short_name,
+                 parse_mode=None,
+                 disable_web_page_preview=None,
```
Comment (jh0ker) [inline telegram/bot.py]: According to https://core.telegram.org/bots/api#sendgame the kwargs should be different

File: telegram/bot.py [hunk 2]
```diff
@@ -1425,6 +1487,81 @@ def getChatMember(self, chat_id, user_id, **kwargs):
...
+            data['chat_id'] = chat_id
+        if message_id:
+            data['message_id'] = message_id
+        if inline_message_id:
+            data['inline_message_id'] = inline_message_id
+        if edit_message:
+            data['edit_message'] = edit_message
+
+        result = self._request.post(url, data, timeout=kwargs.get('timeout'))
+        return Message.de_json(result, self)
```
Comment (jh0ker) [inline telegram/bot.py]: `result` could be just `True` (https://core.telegram.org/bots/api#setgamescore), won't this raise an error?

File: telegram/callbackquery.py [hunk 3]
```diff
@@ -25,14 +25,16 @@
...
         # Required
         self.id = id
         self.from_user = from_user
-        self.data = data
+        self.chat_instance = chat_instance
         # Optionals
         self.message = kwargs.get('message')
+        self.data = kwargs.get('data', '')
         self.inline_message_id = kwargs.get('inline_message_id', '')
+        self.game_short_name = kwargs.get('game_short_name')
```
Comment (jh0ker) [inline telegram/callbackquery.py]: Use an empty string as default

File: telegram/chat.py [hunk 4]
```diff
@@ -57,6 +58,7 @@ def __init__(self, id, type, bot=None, **kwargs):
         self.username = kwargs.get('username', '')
         self.first_name = kwargs.get('first_name', '')
         self.last_name = kwargs.get('last_name', '')
+        self.all_members_are_admins = kwargs.get('all_members_are_admins', '')
```
Comment (jh0ker) [inline telegram/chat.py]: Should be `False` as default, not empty string

File: telegram/game.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        Returns:
+            dict:
+        """
+        data = super(Game, self).to_dict()
+
+        data['photo'] = [p.to_dict() for p in self.photo]
+        data['text_entities'] = [x.to_dict() for x in self.text_entities]
+
+        return data
+
```
Comment (jh0ker) [inline telegram/game.py]: are these just copy-pasted from `telegram.Message`?

File: telegram/gamehighscore.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains a object that represents a Telegram GameHighScore."""
```
Comment (jh0ker) [inline telegram/gamehighscore.py]: an*

File: telegram/game.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        animation (Optional[:class:`telegram.Animation`]): Animation that will be displayed in the
+            game message in chats. Upload via BotFather.
+
+    """
+
+    def __init__(self,
+                 title,
+                 description,
+                 photo,
+                 text=None,
```
Comment (jh0ker) [inline telegram/game.py]: empty string?

File: telegram/webhookinfo.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+    """
+
+    def __init__(self, url, has_custom_certificate, pending_update_count, **kwargs):
+        # Required
+        self.url = url
+        self.has_custom_certificate = has_custom_certificate
+        self.pending_update_count = pending_update_count
+        self.last_error_date = kwargs.get('last_error_date')
+        self.last_error_message = kwargs.get('last_error_message')
```
Comment (jh0ker) [inline telegram/webhookinfo.py]: empty string

File: telegram/game.py (hunk 5 above)
Comment (bomjacob) [inline telegram/game.py]: Yeah, they are. Is there a better way to do it?

File: telegram/gamehighscore.py (hunk 6 above)
Comment (lufte) [inline telegram/gamehighscore.py]: Not meaning to meddle :), but he's most likely talking about "a object", not "a Telegram".

Comment (bomjacob) [inline telegram/gamehighscore.py]: Ah of cause, sorry, I derped. Will fix shortly :)

File: telegram/game.py (hunk 5 above)
Comment (tsnoam) [inline telegram/game.py]: @bomjacob 
I think what @jh0ker means is that you don't need to manually serialize a list. the base class will do it for you. (and yes, i missed it myself going over the code)

**Pull request merged**